From 8085c2b984707640eac8f56680a451a598488c52 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Mon, 8 Jun 2020 11:01:24 +0300
Subject: [PATCH 54/55] ARM: dts: imx7d: cl-som-imx7: rework USB hub reset

Rework USB hub reset. The hub is connected to USB HSIC controller.
Use regulator subsystem instead of gpio-hog.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 arch/arm/boot/dts/imx7d-cl-som-imx7.dts | 18 ++++++++++--------
 1 file changed, 10 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-cl-som-imx7.dts b/arch/arm/boot/dts/imx7d-cl-som-imx7.dts
index 922c015ed371..277a35cc4f89 100644
--- a/arch/arm/boot/dts/imx7d-cl-som-imx7.dts
+++ b/arch/arm/boot/dts/imx7d-cl-som-imx7.dts
@@ -36,6 +36,16 @@
 		enable-active-high;
 	};
 
+	reg_usbh_nreset: regulator-usbh-nrst {
+		compatible = "regulator-fixed";
+		regulator-name = "usb_host_nreset";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		gpio = <&pca9555 6 GPIO_ACTIVE_LOW>;
+		enable-active-high;
+		regulator-always-on;
+	};
+
 	tsc2046reg: regulator-tsc {
 		compatible = "regulator-fixed";
 		regulator-name = "tsc2046-reg";
@@ -363,14 +373,6 @@
 		#gpio-cells = <2>;
 		reg = <0x20>;
 
-		/* USB Hub nRST */
-		usb_hub_nreset {
-			gpio-hog;
-			gpios = <6 GPIO_ACTIVE_HIGH>;
-			output-high;
-			line-name = "usb_hub_nreset";
-		};
-
 		/* WLAN RF Power enable GPIO */
 		rf_pwr_en_reg {
 			gpio-hog;
-- 
2.20.1

